tasks/new_e2e_tests.py

```diff
@@ -194,6 +196,49 @@ def clean(ctx, locks=True, stacks=False, output=False, skip_destroy=False):
         _clean_output()
 
 
+@task
+def cleanup_remote_stacks(ctx, pipeline_id, pulumi_backend):
+    """
+    Clean up remote stacks created by the pipeline
+    """
+    # if not running_in_ci():
+    #     raise Exit("This task should be run in CI only", 1)
+    print(f"Running aws s3api list-objects --bucket {pulumi_backend} --prefix .pulumi/stacks/e2eci/ci-{pipeline_id}")
+    res = ctx.run(
+        f"aws s3api list-objects --bucket {pulumi_backend} --prefix .pulumi/stacks/e2eci/ci-{pipeline_id}",
+        hide=True,
+        warn=True,
+    )
+    if res.exited != 0:
+        print(f"Failed to list stacks in bucket {pulumi_backend}:", res.stdout, res.stderr)
+        return
+    eks_stacks = set()
+    stacks = json.loads(res.stdout)
+    if "Contents" in stacks:
+        for stack in stacks["Contents"]:
+            if "Key" in stack:
+                stack_id = stack["Key"].split("/")[-1].replace(".json.bak", "").replace(".json", "")
```

Comment:
```suggestion
    for stack in stacks.get("Contents", []):
        stack_id = stack.get("Key", "").split("/")[-1].replace(".json.bak", "").replace(".json", "")
```
Suggestion to have the code less verbose